pkg/utils/common.go

```diff
@@ -36,3 +37,29 @@ func SliceFilter[T any](in []T, f FilterFunc[T]) []T {
 func LatticeServiceName(name string, namespace string) string {
 	return fmt.Sprintf("%s-%s", Truncate(name, 20), Truncate(namespace, 18))
 }
+
+func PointerStringSlicesElementsEqual(arr1, arr2 []*string) bool {
+	if len(arr1) != len(arr2) {
+		return false
+	}
+
+	map1 := make(map[string]int)
+	map2 := make(map[string]int)
+
+	for _, str := range arr1 {
+		if str != nil {
+			map1[*str]++
+		} else {
+			map1["<nil>"]++
+		}
+	}
+
+	for _, str := range arr2 {
+		if str != nil {
+			map2[*str]++
+		} else {
+			map2["<nil>"]++
```

Comment:
is this necessary if the `str` is `nil`? Since you compared the length of the 2 slices anyway. You can just not populate `map1` and `map2` if `str` is nil

Reply:
Also add unit tests for this method